src/main/java/org/jboss/ejb/protocol/remote/EJBServerChannel.java

```diff
@@ -325,8 +327,13 @@ private void handleTxnRequest(final int code, final int invId, final MessageInpu
                     }
                     default: throw Assert.impossibleSwitchCase(code);
                 }
-            } catch (SystemException | HeuristicMixedException | RollbackException | HeuristicRollbackException e) {
-                writeFailedResponse(invId, e);
+            } catch (Throwable t) {
+                // Narayana uses Errors, Exceptions, and RuntimeExceptions
+                if (t instanceof VirtualMachineError) {
```

Comment:
In this case we should still write a failed response.  It should be a SystemException.